syncstorage/src/tokenserver/auth/oauth.rs

```diff
@@ -112,31 +139,41 @@ impl VerifyToken for RemoteVerifier {
                     ..TokenserverError::invalid_credentials("Unauthorized")
                 }),
             }
-        })
-        .map_err(|err| {
-            let context = if err.is_cancelled() {
-                "Tokenserver threadpool operation cancelled"
-            } else if err.is_panic() {
-                "Tokenserver threadpool operation panicked"
-            } else {
-                "Tokenserver threadpool operation failed for unknown reason"
-            };
+        };
+
+        if self.jwk_is_cached {
+            verify_inner(self)
+        } else {
+            let verifier = self.clone();
+
+            // If the JWK is not cached, PyFxA will make a request to FxA to retrieve it, blocking
+            // this thread. To improve performance, we make the request on a thread in a threadpool
+            // specifically used for blocking operations.
+            let fut = task::spawn_blocking(move || verify_inner(&verifier)).map_err(|err| {
```

Comment:
There are many more threads in tokio's blocking threadpool than there are CPU cores, so spawning the task on a blocking thread even when verifying locally could lead to performance degradation.